### DISCUSSION TITLE
"No function matches the given name and argument types. You might need to add explicit type casts."

### DISCUSSION THREAD

**Discussion by prescience-data** [supabase/supabase#2975] [Questions] [answered]
Getting this odd error, despite having many other functions working:
```json
{
 "code": "42883",
 "details": null,
 "hint": "No function matches the given name and argument types. You might need to add explicit type casts.",
 "message": "function public.fetch_account() does not exist"
}
```

```pgsql
-- CreateFunction
CREATE OR REPLACE FUNCTION fetch_account
(
)
  RETURNS jsonb
  LANGUAGE plpgsql
AS
$$
    # VARIABLE_CONFLICT USE_COLUMN
BEGIN
 -- // Function logic
END;
$$;

-- GrantExecute
GRANT EXECUTE ON FUNCTION fetch_account() TO authenticated;
```

```typescript

const { data, error } = await client().rpc("fetch_account").single()

```

I have a similar error with another new function I just created which has a `bigint` param  `fetch_team( id bigint )`  - I thought it was an issue with the param, however, this parameterless function seems like it shouldn't have any ambiguity?

**Comment by soedirgo** [comment]
I wasn't able to reproduce this:
```sql
CREATE OR REPLACE FUNCTION fetch_account
(
)
  RETURNS jsonb
  LANGUAGE plpgsql
AS
$$
BEGIN
  RETURN '{}'::jsonb;
END;
$$;
```
```js
const { createClient } = require('@supabase/supabase-js')
const supabase = createClient(...,  ...);
supabase.rpc('fetch_account').then(console.log)
```
```js
{
  error: null,
  data: {},
  count: null,
  status: 200,
  statusText: 'OK',
  body: {}
}
```

Can you see your function in Database > Alpha preview > Functions?

**Reply by prescience-data** [reply]
hmm interesting... neither of the two problem functions are showing in that screen?
I've never touched (or seen) that screen before (last time I added functions was ~2 months ago from memory) - is there some extra step required now?

(to clarify, I just added these two new functions to my migrations today)

**Reply by soedirgo** [reply]
Can you execute the function from the SQL editor? Just checking if you created the function in a different project.

**Reply by prescience-data** [reply]
Yes, part of the migration script is executing each function as a test.
It works when running from the SQL editor, despite not showing in that screen. 🤔

**Comment by steve-chavez** [comment, marked as answer]
@prescience-data Can you confirm that `fetch_account` is inside the `public` schema?

Try executing:

```sql
SELECT specific_schema
FROM information_schema.routines
WHERE routine_name='fetch_account';
```

It should show the schema of the function.

**Reply by prescience-data** [reply]
```
RUN
Success. No rows returned
```

This is even with adding `public.` as a prefix: 

```sql

-- CreateFunction
CREATE OR REPLACE FUNCTION public.fetch_account
```

**Reply by prescience-data** [reply]
Further, when I run:
```sql
-- Test
SELECT *
FROM fetch_account( );
-- Test
SELECT *
FROM fetch_team( 1 );
```
at the end of the migration SQL script, it returns successfully. 

However, in a new SQL tab, if I run it, I receive:
```
function fetch_account() does not exist
function fetch_team(integer) does not exist
```

**Reply by steve-chavez** [reply]
> RUN
> Success. No rows returned

@prescience-data The above means that the function doesn't exist in the database.

> function fetch_account() does not exist
> function fetch_team(integer) does not exist

Which is confirmed by the above

> at the end of the migration SQL script

I guess your migration script is not working somehow(or not committing). Maybe try to create the function manually on the SQL Editor and retry.

**Reply by prescience-data** [reply]
> your migration script is not working somehow(or not committing)

I believe this must have been the case, despite no errors being thrown in the console.
I'm trying to work out how that could occur though, given final stage of the script runs a bunch of INSERTs, and those rows were consistently present (and timestamped) when viewing the data directly.

**Comment by w3b6x9** [comment]
@prescience-data can you please email [support@supabase.io](support@supabase.io) with your project ref so that we may investigate further?
